### ISSUE TITLE
Forwarding XFCC header to route-service.

### ISSUE THREAD

**Issue by mandar14** [cloudfoundry/routing-release#95]
## Issue

Client Certificate received in a request are no getting forwarded to route-service via router.

## Context

After setting newly introduced property `router.forwarded_client_cert` found that,
XFCC header can be found in request headers sent to an app directly

```
client --(cert)--> Router--(XFCC header)--> end_app
```
But gorouter don't set XFCC header when proxy-ing request via routing-service attached to an application.

client --(cert)--> Router--(_**no XFCC header**_)--> Route-service---> end_app

## Steps to Reproduce
Router configuration (using v0.168.0)
```
router:
  client_cert_validation: require
  disable_http: true
  enable_ssl: true
  forwarded_client_cert: sanitize_set
  min_tls_version: TLSv1.1
  ssl_skip_validation: false
```

1. Push demo_app list all header on request
2. Push and bind any route-service and to demo app. List headers coming at **route-service** after requesting **demo_app**.

It seems router is setting following 3 headers only while forwarding request to route-service.  Probably conditional XFCC header need to be added.
https://godoc.org/code.cloudfoundry.org/gorouter/routeservice#pkg-constants

Thanks,
Mandar K.

**Comment by shalako** [comment]
I reproduced the issue. We'll investigate changing the behavior.

**Comment by nmahoney-pivotal** [comment]
Hi @mandar14 ,

We've investigated this issue and believe we've reproduced it and found a root cause.  However, we used a slightly different setup than you, so we want to confirm a few things:

There are two relevant parameters:
- `disable_http` is `true` or `false`
- the route service is backed by an App on CF or an External HTTP server

1) When we try with `disable_http: true` with the Route Service running as an App on CF, then we found that the route service was completely broken, and the Go Router responded to the client with a 502 error.  We believe we've identified the root cause of this, relating to some tricky loopback behavior in the gorouter ([bug here](https://www.pivotaltracker.com/story/show/154257824)).

2) With `disable_http: false` and the Route Service running as an App on CF, then we see the behavior you described: the XFCC header gets stripped from the request so the route service never sees it.  

Could you confirm whether you had `disable_http` set to `true` or `false`?  We want to make sure we've reproduced your actual issue.

With an external route service, the `disable_http` flag shouldn't matter, but we also expect that the XFCC header should arrive at the route service.

**Comment by mandar14** [comment]
Yes disable _http was set to true. Didn't tried external route service is there any available for authentication (certificate based)

**Comment by shubhaat** [comment]
We have fixed this issue with routing-release 0.176.0.
